### PR TITLE
networkmanagerapplet: 1.18.0 -> 1.20.0

### PR DESCRIPTION
**Pull request by amaxine**
###### Motivation for this change
Version bump.

Notably allows disabling/enabling wireguard connections, instead of just editing them (though until NM gets updated this is meaningless). Only tested in as far as running it in `--indicator` mode.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by maxeaubrey** [comment]
Result of `nixpkgs-review pr 115415` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>gnome3.gnome-control-center</li>
    <li>gnome3.networkmanagerapplet</li>
    <li>pantheon.elementary-greeter</li>
    <li>pantheon.switchboard-plug-network</li>
    <li>pantheon.switchboard-with-plugs</li>
    <li>pantheon.wingpanel-applications-menu</li>
    <li>pantheon.wingpanel-with-indicators</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 115415` at 4957eaf7 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/d0c134a14b58f26117c52abeed9a8de3">gnome3.gnome-control-center</a></li> (gnome3.gnome_control_center)</li>
    <li><a href="https://gist.github.com/e8b5c97e153b38dd971a5f6dbf39de61">networkmanagerapplet</a></li> (gnome3.networkmanagerapplet)</li>
    <li><a href="https://gist.github.com/3736a2a1fc0fe771ec413a39d9ed066e">pantheon.elementary-greeter</a></li></li>
    <li><a href="https://gist.github.com/d3e07a3d0a9d36341b8cbec3e09f25a7">pantheon.switchboard-plug-network</a></li></li>
    <li><a href="https://gist.github.com/5a8cdfb7b62c3ca658dfe36a22e42baa">pantheon.switchboard-with-plugs</a></li></li>
    <li><a href="https://gist.github.com/82efd6503112031cd0a4a678dec7ce40">pantheon.wingpanel-applications-menu</a></li></li>
    <li><a href="https://gist.github.com/77201dc41ceaa7df791768acc27d6025">pantheon.wingpanel-with-indicators</a></li></li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/networking/network-manager/applet/default.nix:84:5:
```
   |
84 |     license = licenses.gpl2;
   |     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `glib`.
Near pkgs/tools/networking/network-manager/applet/default.nix:24:3:
```
   |
24 | , glib
   |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `mobile-broadband-provider-info`.
Near pkgs/tools/networking/network-manager/applet/default.nix:14:3:
```
   |
14 | , mobile-broadband-provider-info
   |   ^
```
</li>
  </ul>
</details>

**Comment by maxeaubrey** [comment]
The need for `glib` and `mobile-broadband-provider-info` was dropped in 50438bd5b1589dfd3a4dfa407b81441ed42fc8c7. I think gpl2Only is correct here.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115415` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
7 packages built:
</summary>
<ul>
<li>gnome3.gnome-control-center (gnome3.gnome_control_center)</li>
<li>networkmanagerapplet (gnome3.networkmanagerapplet)</li>
<li>pantheon.elementary-greeter</li>
<li>pantheon.switchboard-plug-network</li>
<li>pantheon.switchboard-with-plugs</li>
<li>pantheon.wingpanel-applications-menu</li>
<li>pantheon.wingpanel-with-indicators</li>
</ul>
</details>

File: pkgs/tools/networking/network-manager/applet/default.nix [hunk 1]
```diff
@@ -81,7 +79,7 @@ stdenv.mkDerivation rec {
   meta = with lib; {
     homepage = "https://gitlab.gnome.org/GNOME/network-manager-applet/";
     description = "NetworkManager control applet for GNOME";
-    license = licenses.gpl2;
+    license = licenses.gpl2Only;
```
Comment (jtojnar) [inline pkgs/tools/networking/network-manager/applet/default.nix:82]: It is necessary to look into the source code: https://gitlab.gnome.org/GNOME/network-manager-applet/-/blob/74a71cbe52c961701d4d8b70bcf76fa0ed83a5d3/meson.build#L4

The license identified by GitHub/GitLab is misleading. See also https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206.

File: pkgs/tools/networking/network-manager/applet/default.nix [hunk 2]
```diff
@@ -21,16 +20,15 @@
 , python3
 , gtk3
 , libappindicator-gtk3
-, glib
```
Comment (jtojnar) [inline pkgs/tools/networking/network-manager/applet/default.nix:24]: The code does in fact depend on GLib:

https://gitlab.gnome.org/GNOME/network-manager-applet/-/blob/74a71cbe52c961701d4d8b70bcf76fa0ed83a5d3/meson.build#L107

(It gets it from GTK propagated build inputs but we generally prefer listing direct dependencies explicitly.)

Comment (maxeaubrey) [reply pkgs/tools/networking/network-manager/applet/default.nix:24]: Ack, thanks for noting.

File: pkgs/tools/networking/network-manager/applet/default.nix (hunk 1 above)
Comment (maxeaubrey) [reply pkgs/tools/networking/network-manager/applet/default.nix]: Ack, thanks for noting.

File: pkgs/tools/networking/network-manager/applet/default.nix (hunk 2 above)
Comment (maxeaubrey) [reply pkgs/tools/networking/network-manager/applet/default.nix:24]: So, it should have it listed in `buildInputs` as well, then?

Comment (jtojnar) [reply pkgs/tools/networking/network-manager/applet/default.nix:24]: Yup.

**Comment by jtojnar** [comment]
@GrahamcOfBorg build network-manager-applet

**Review by jtojnar — APPROVED** [review]
Looks good. Thanks.

**Comment by jtojnar** [comment]
Actually, the commit message should say `networkmanagerapplet` so that ofborg builds it.

**Comment by maxeaubrey** [comment]
> Actually, the commit message should say networkmanagerapplet so that ofborg builds it.

Fixed. The `pname` left me confused.

**Comment by jtojnar** [comment]
Yeah, I guess we might want to rename the attribute eventually to be consistent with upstream name.

**Pull request merged**